File: java/plzrun/Boj/src/dp/Boj1912.java
package dp;
import java.util.*;

public class Boj1912 {
	static int[] cash;
	static int[] dp;
	
	public static void main(String[] args) {
		Scanner scanner=new Scanner(System.in);
		
		int number;
		int max=Integer.MIN_VALUE;
		number=scanner.nextInt();
		
		cash=new int[number+1];
		dp=new int[number+1];
		
		for(int i=1; i<=number; i++) {
			cash[i]=scanner.nextInt();
		}
		
		dp[0]=0;
		
		if(number>=1)
			for(int i=1; i<=number; i++) {
				dp[i]=cash[i];
				if(0<dp[i-1])
					dp[i]=dp[i-1]+cash[i];
			}
		
		for(int i=1; i<=number; i++)
			max=Math.max(max, dp[i]);
		
		System.out.println(max);
		
		
		scanner.close();
		return;
	}
}

File: c++/plzrun/DpCpp/Boj1912/Boj1912.cpp
#include <iostream>

using namespace std;

int input[100001];
int dp[100001];

int main() {
	int number;
	int max;
	dp[0] = 0;

	cin >> number;

	for (int i = 1; i <= number; i++) {
		cin >> input[i];
		if (i == 1)
			max = input[i];

		dp[i] = input[i];
		if (dp[i - 1] > 0)
			dp[i] += dp[i-1];

		if (max < dp[i])
			max = dp[i];
	}

	cout << max;

	return 0;
}